_posts/2022-05-18-My-OTP-25-highlights.md

```diff
@@ -0,0 +1,751 @@
+* [New functions in the `maps`and `lists` modules](#new-functions-in-the-mapsand-lists-modules)
+* [Selectable features and the new `maybe_expr` feature](#selectable-features-and-the-new-maybe_expr-feature)
+* Compiler News
+* Crypto and OpenSSL 3.0
+* Dialyzer
+* [Improvements of the JIT](#improvements-of-the-jit)
+* ETS-tables with adaptive support for write concurrency
+* Relocatable installation directory for Erlang
+* New option `short` to the functions `erlang:float_to_list/2` and `erlang:float_to_binary/2` 
+* The new module `peer` supersedes the slave module
+* global will now by default prevent overlapping partitions
+* gen_server, gen_statem and gen_event has got a new format_status/1 callback.
+* The `timer` module has been modernized and made more efficient
+* CA-certificates can be fetched from the OS standard place 
+
+
+# New functions in the `maps`and `lists` modules
+
+Triggered by suggestions from the users we have introduced new functions in the `maps` and `lists` modules in `stdlib.
+
+## `maps:groups_from_list/2,3`
+
+For short we can say that this function take a list of elements and group them. The result is a map `#{Group1 => [Group1Elements], GroupN => [GroupNElements]}`.
+
+Let us look at some examples from the shell:
+
+```erlang
+> maps:groups_from_list(fun(X) -> X rem 2 end, [1,2,3]).
+#{0 => [2], 1 => [1, 3]}
+```
+The provided fun calculates X rem 2 for every element X in the input list and then group the elements in a map with the result of X rem 2 as key and the corresponding elements as a list for that key.
+
+```erlang
+> maps:groups_from_list(fun erlang:length/1, ["ant", "buffalo", "cat", "dingo"]).
+#{3 => ["ant", "cat"], 5 => ["dingo"], 7 => ["buffalo"]}
+```
+
+In the example above the strings in the input list are grouped into a map based on their length.
+
+## `maps:groups_from_list/3`
+
+There is also a variant of groups_from_list with an additional fun by which the values can be converted or changed before they are put into their groups.
+
+```erlang
+> maps:groups_from_list(fun(X) -> X rem 2 end, fun(X) -> X*X end, [1,2,3]).
+#{0 => [4], 1 => [1, 9]}
+```
+In the example above the elements X in the list are grouped according the X rem 2 calculation but the values stored in the groups are the elements multiplied by them selves (X*X).  
+
+```erlang
+> maps:groups_from_list(fun erlang:length/1, fun lists:reverse/1, ["ant", "buffalo", "cat", "dingo"]).
+#{3 => ["tna","tac"],5 => ["ognid"],7 => ["olaffub"]}
+```
+
+In the example above the strings from the input list are grouped according to their length and they are reversed before they are stored in the groups.
+
+## `lists:enumerate/1,2`
+
+Takes a list of elements and return a new list where each element has been associated with its position in the original list. Returns a new list with tuples of the form `{I, H}` where `I` is the position of `H` in the original list. The enumeration starts with 1 and increases by 1 in each step.
```

Comment:
"Takes a list of elements and return ..."  'Takes' is 3rd person singular but 'return' is not although same subject.  It should it be 'return**s**'